upup/pkg/fi/cloudup/openstacktasks/instance.go

```diff
@@ -97,30 +100,55 @@ func (e *Instance) Find(c *fi.Context) (*Instance, error) {
 	if e == nil || e.Name == nil {
 		return nil, nil
 	}
+	tmp := strings.Replace(strings.ToLower(fi.StringValue(e.GroupName)), "_", "-", -1)
+	groupName := strings.Replace(tmp, ".", "-", -1)
 	serverPage, err := servers.List(c.Cloud.(openstack.OpenstackCloud).ComputeClient(), servers.ListOpts{
-		Name: fmt.Sprintf("^%s$", fi.StringValue(e.Name)),
+		Name: fmt.Sprintf("^%s", groupName),
 	}).AllPages()
 	if err != nil {
-		return nil, fmt.Errorf("error finding server with name %s: %v", fi.StringValue(e.Name), err)
+		return nil, fmt.Errorf("error listing servers: %v", err)
 	}
 	serverList, err := servers.ExtractServers(serverPage)
 	if err != nil {
 		return nil, fmt.Errorf("error extracting server page: %v", err)
 	}
-	if len(serverList) == 0 {
+
+	filteredList := []servers.Server{}
+	for _, server := range serverList {
+		val, ok := server.Metadata["k8s"]
+		if !ok || val != fi.StringValue(e.ClusterName) {
```

Comment:
The "generic" way to do this would be to pass in `filterTags map[string]string`.  Or just `tags map[string]string` if they are used both for filtering and creation.

An advantage of that is that I'm not sure that these values are also set on creation.  I'm sure they are, but ... it would be more obvious if it was just `tags map[string]string` :-)